splink/input_column.py

```diff
@@ -118,135 +159,171 @@ def parse_input_name_to_sqlglot_tree(self) -> Expression:
-            return tree
-
-    def from_settings_obj_else_default(self, key, schema_key=None):
-        # Covers the case where no settings obj is set on the comparison level
-        if self._settings_obj:
-            return getattr(self._settings_obj, key)
-        else:
-            if not schema_key:
-                schema_key = key
-            return default_value_from_schema(schema_key, "root")
+            return identifier, None
 
     @property
-    def gamma_prefix(self) -> str:
-        return self.from_settings_obj_else_default(
-            "_gamma_prefix", "comparison_vector_value_column_prefix"
-        )
-
-    @property
-    def bf_prefix(self) -> str:
+    def _bf_prefix(self):
         return self.from_settings_obj_else_default(
             "_bf_prefix", "bayes_factor_column_prefix"
         )
 
     @property
-    def tf_prefix(self) -> str:
+    def _tf_prefix(self):
         return self.from_settings_obj_else_default(
             "_tf_prefix", "term_frequency_adjustment_column_prefix"
         )
 
+    def _copy_with_new_column_builder(self, new_builder) -> InputColumn:
+        input_column_copy = deepcopy(self)
+        input_column_copy.column_tree_builder = new_builder
+        return input_column_copy
+
+    def change_input_column_name(self, new_name: str):
+        return self._copy_with_new_column_builder(
+            self.column_tree_builder.change_name(new_name)
+        )
+
+    def unquote(self) -> InputColumn:
+        return self._copy_with_new_column_builder(self.column_tree_builder.unquote())
+
     @property
-    def name(self) -> str:
-        return self.input_name_as_tree.sql(dialect=self._sql_dialect)
+    def as_base_dialect(self) -> InputColumn:
+        input_column_copy = deepcopy(self)
+        input_column_copy.sqlglot_name = None
+        return input_column_copy
+
+    def column_tree_to_sql(self, column_tree: ColumnTreeBuilder) -> str:
+        return column_tree.build_column_tree().sql(self.sqlglot_name)
+
+    def _table_name_as(self, table: str, prefix: str = "", suffix: str = "") -> str:
+        # Remove quotes, if applied. This prevents
+        # the accidental addition of quotes to the alias.
+        # See: exp.Identifier(this='`test`', quoted=True).sql("spark")
+        column_builder = self.column_tree_builder.unquote()
+        # Add both the prefix and suffix arguments to the column
+        alias_tree = column_builder.add_prefix(prefix).add_suffix(suffix)
+        alias_string = self.column_tree_to_sql(alias_tree)
+
+        column_with_table = self.column_tree_builder.add_table(table).add_prefix(prefix)
+        tree = column_with_table.add_alias(alias_string)
+        return tree.sql(self.sqlglot_name)
 
     @property
-    def name_l(self) -> str:
-        return add_suffix(self.input_name_as_tree, suffix="_l").sql(
-            dialect=self._sql_dialect
-        )
+    def name(self):
+        tree = self.column_tree_builder
+        return self.column_tree_to_sql(tree)
```

Comment:
if the `tree` knew about its dialect, then it could have a `.sql` property and this could be
```
return self.column_tree_builder.sql
```

Reply:
My original implementation had the dialect embedded inside the class. I ended up dropping that to simplify it.

Given the complexity that results from getting the InputColumn to do the `.sql()...` work, I think you're right and I'll migrate back to the old structure.

Reply:
I think part of my wanted to uphold the notion that the `ColumnTreeBuilder` was merely a "tree builder". 

However, complicates the code more than is necessary.